### ISSUE TITLE
dirpath built in preprocessor function returns empty in pipe mode

### ISSUE THREAD

**Issue by Fruchtzwerg94** [plantuml/plantuml#1910]
**Bug description**
According to the [CLI documentation](https://plantuml.com/command-line), in pipe mode, setting 
`-filename "file.puml"           Override %filename% variable`
allows to set the `%filename()` preprocessor function. Similar to that,
`-filedir xxx                    Behave as if the PlantUML source is in this dir (only affects '-pipe' and PicoWeb 'POST /render')`
can be set, but this has no effect on the `%dirpath()` preprocessor function, it returns empty.

**To Reproduce**
Run PlantUML in pipe mode with CLI arguments
`-pipe -charset UTF-8 -tsvg -stdrpt:1 -filedir "C:\Users\user\Desktop\test" -filename "new 1.txt"`
Pipe in the diagram:
```
@startuml
note: %filename()
note: %dirpath()
@enduml
```

**Expected behavior**
`%filename()` and `%dirpath()` can be resolved.

**Screenshots**
![grafik](https://github.com/user-attachments/assets/3b6b0f22-50c1-4373-9c51-a9b6ddd0ba48)

**Comment by The-Lum** [comment]
Hello @Fruchtzwerg94 

FYI:

> Starting with version 1.2020.11, we fix some security issues in PlantUML (see https://plantuml.com/security )
>
> In some occasions, it turns out that %dirpath() exposes too many info about the running server.
> In your case, you have to set PLANTUML_SECURITY_PROFILE to UNSECURE to let %dirpath() working.
>
> Tell us if it help !


Ref.:
- https://forum.plantuml.net/12796/%25dirpath-is-not-working-after-1-2020-10?show=12799#a12799

_If that can help,_
Regards,
Th.

**Comment by Fruchtzwerg94** [comment]
Hi @The-Lum ,

thanks for the information. Unfortunately setting the profile to `UNSECURE` does not change the outcome.

```
java.exe
    -Dfile.encoding=UTF-8
    -DPLANTUML_SECURITY_PROFILE=UNSECURE
    -jar "plantuml-1.2024.7.jar"
    -pipe
    -charset UTF-8
    -tsvg
    -stdrpt:1
    -filedir "C:\Users\user\Desktop\test"
    -filename "new 1.txt"
```
Anything else you see here what may missed?

BR, Philipp

**Comment by The-Lum** [comment]
Hi @Fruchtzwerg94 

See perhaps other elements here:
- https://forum.plantuml.net/10199/%25dirpath-returns-empty-string-on-proxy?show=10202#a10202

If that can help,
Regards,
Th.

**Comment by Fruchtzwerg94** [comment]
Hi @The-Lum ,

from what is mentioned there
> When you are using PlantUML in cloud/internet context, the underlying library cannot exactly guess who is calling it.
 and what I've found in other discussions the pipe mode should not be affected by this constraint. But my feeling is that this is not perfectly documented and pipe mode is affected here as well?

BR,
Philipp

**Comment by The-Lum** [comment]
Hi @Fruchtzwerg94 

Here is a workaround, with a `DIRPATH` variable definition, as:
```sh
java.exe
    -Dfile.encoding=UTF-8
    -DPLANTUML_SECURITY_PROFILE=UNSECURE
    -jar "plantuml-1.2024.7.jar"
    -pipe
    -charset UTF-8
    -tsvg
    -stdrpt:1
    -filedir "C:\Users\user\Desktop\test"
    -filename "new 1.txt"
    -DDIRPATH="C:\Users\user\Desktop\test"
```

and:
```plantuml
@startuml
note: %filename()
note: DIRPATH
@enduml
```


Then:
- What is your usecase?
- Why do you want `dirpath` on pipe mode?

_If that can help,
Enjoy,_
Regards,
Th.

**Comment by Fruchtzwerg94** [comment]
Hi @Fruchtzwerg94 ,

I am working on the [PlantUmlViewer](https://github.com/Fruchtzwerg94/PlantUmlViewer) Notepad++ plugin where PlanUML is called in pipe mode. To allow usage of PlantUML in its default behaviour `%dirpath()` shall be supported as well (like possible with `%filename()`). Since many users are expecting the default behaviour, defining a custom variable seems not to be a solution here.

BR,
Philipp

**Comment by The-Lum** [comment]
> Since many users are expecting the default behaviour, defining a custom variable seems not to be a solution here.

_I will let @arnaudroques answer you who will be better able to answer you on this `pipe` subject._
Regards,
Th.

**Comment by The-Lum** [comment]
Hi all,

This is now fixed, on [v1.2024.8](https://github.com/plantuml/plantuml/releases/tag/v1.2024.8)
- Thanks @Fruchtzwerg94 for the report.
- Thanks PlantUML team for your work.

Regards.

**Comment by Fruchtzwerg94** [comment]
Nice, its now also released with the latest version of [PlantUmlViewer](https://github.com/Fruchtzwerg94/PlantUmlViewer).
Thanks!
